src/lib/y2partitioner/widgets/disk_table.rb

```diff
@@ -0,0 +1,93 @@
+require "yast"
+
+require "cwm/table"
+
+require "y2partitioner/icons"
+require "y2partitioner/widgets/blk_devices_table"
+
+module Y2Partitioner
+  module Widgets
+    # Table widget to represent given list of Y2Storage::BlkDevice.
+    class DiskTable < CWM::Table
+      include BlkDevicesTable
+
+      # @param disks [Array<Y2Storage::Disk|Y2Storage::Partition>] devices to display
+      # @param pager [CWM::Pager] table have feature, that double click change content of pager
+      #   if someone do not need this feature, make it only optional
+      def initialize(disks, pager)
+        textdomain "storage"
+        @disks = disks
+        @pager = pager
+      end
+
+      # table items. See CWM::Table#items
+      def items
+        @disks.map do |device|
+          [
+            id_for_device(device), # use name as id
+            device.name,
+            device.size.to_human_string,
+            # TODO: dasd format use "X", check it
+            device.exists_in_probed? ? "" : _(BlkDevicesTable::FORMAT_FLAG),
```

Comment:
Assuming, of course, that "formatting" means literally that (creating a filesystem). In case you also want to return `true` for devices that will be used as PV or stuff like that, you should adapt the method according.